bindings/node/lib/mongocryptdManager.js

```diff
@@ -0,0 +1,123 @@
+'use strict';
+
+const spawn = require('child_process').spawn;
+const readFile = require('fs').readFile;
+
+/**
+ * @typedef AutoEncryptionExtraOptions
+ * @prop {string} [mongocryptdURI] overrides the uri used to connect to mongocryptd
+ * @prop {boolean} [mongocryptdBypassSpawn=false] if true, autoEncryption will not spawn a mongocryptd
+ * @prop {string} [mongocryptdSpawnPath] the path to the mongocryptd executable
+ * @prop {string[]} [mongocryptdURI] command line arguments to pass to the mongocryptd executable
+ */
+
+module.exports = function() {
+  const platform = require('os').platform;
+
+  const MONGOCRYPTD_PID_FILE = 'mongocryptd.pid';
+  const STATE = {
+    NO_PID_FILE: Symbol('NO_PID_FILE'),
+    EMPTY_PID_FILE: Symbol('EMPTY_PID_FILE'),
+    VALID_PID_FILE: Symbol('VALID_PID_FILE')
+  };
+
+  const checkIntervalMS = 100;
+
+  function checkPidFile(callback) {
+    readFile(MONGOCRYPTD_PID_FILE, 'utf8', (err, data) => {
+      if (err) {
+        return callback(undefined, STATE.NO_PID_FILE);
+      }
+
+      if (!data || !data.length) {
+        return callback(undefined, STATE.EMPTY_PID_FILE);
+      }
+
+      try {
+        JSON.parse(data);
+      } catch (e) {
+        return callback(e);
+      }
+
+      callback(undefined, STATE.VALID_PID_FILE);
+    });
+  }
+
+  function waitForPidFile(up, tries, callback) {
+    if (tries <= 0) {
+      return callback();
+    }
+
+    checkPidFile((err, state) => {
+      if ((state === STATE.VALID_PID_FILE) === up) {
+        return callback();
+      }
+
+      tries -= 1;
+      setTimeout(() => waitForPidFile(up, tries, callback), checkIntervalMS);
+    });
+  }
+
+  class MongocryptdManager {
```

Comment:
similarly, this file doesn't seem to need any injection. Should it still follow the pattern?